distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -681,100 +650,103 @@ type Table
-
-            Table.Value new_table_name self.connection new_columns new_ctx
+       - right: The table to join with.
+       - join_kind: The `Join_Kind` for the joining the two tables.
+       - on: A single condition or a common column name, or a list thereof, on
+         which to correlate rows from the two tables. If multiple conditions
+         are supplied, rows are correlated only if all are true.
+         If common column names are provided, these columns should be present
+         in both tables and an equality condition is added for each of them.
+       - right_prefix: The prefix added to right table column names in case of
+         name conflict.
+       - on_problems: Specifies how to handle problems if they occur, reporting
+         them as warnings by default.
+
+         - If a column name cannot be found, a `No_Such_Column` is reported
+           and an empty result is reported.
+         - If a column index is invalid, an `Index_Out_Of_Bounds` is
+           reported and an empty result is reported.
+         - If a join condition correlates columns whose types are not compatible
+           (for example comparing numeric types with text), an
+           `Invalid_Value_Type` is reported.
+         - If decimal columns are joined on equality, a
+           `Floating_Point_Grouping` is reported.
+
+         In any of the above cases, if a problem occurs, the resulting table
+         will have the desired structure, but it will be empty to indicate that
+         the join has failed due to an erroneous join condition.
+
+       ? Column Renaming
+
+         If columns from the two tables have colliding names, a prefix (by
+         default `Right_`) is added to the name of the column from the right
+         table. The left column remains unchanged. It is possible that the new
+         name will be in use, in this case it will be resolved using the normal
+         renaming strategy - adding subsequent `_1`, `_2` etc.
+
+       ? Result Ordering
+
+         The ordering of rows in the resulting table is not specified.
+
+       ? Joining on equality of columns with the same name
+
+         When joining two columns with the same name and an equality condition,
+         only one copy of column will be included in the output (avoiding
+         unnecessary duplication and renaming).
+
+       ? Same-name column join shorthand
+
+         As a shorthand, providing a column name or a list of column names
+         allows to join the two tables on equality of corresponding columns with
+         the same name. So `table.join other on=["A", "B"]` is a shorthand for:
+             table.join other on=[Join_Condition.Equals "A" "A", Join_Condition.Equals "B" "B"]
+    join : Table -> Join_Kind -> Join_Condition | Text | Vector (Join_Condition | Text) -> Text -> Problem_Behavior -> Table
+    join self right join_kind=Join_Kind.Inner on=[Join_Condition.Equals 0 0] right_prefix="Right_" on_problems=Report_Warning =
```

Comment:
should the parameter be called `right`? I'm asking because it will show up on UI so maybe it should just be renamed to `table` thus creating less confusion

Reply:
I think we should keep as `right` as it makes it clear it is the right of a join. Given we have left/right on other parts it aligns clearly.